dashboard/lib/libraries_list.rb

```diff
@@ -0,0 +1,17 @@
+include ProjectsList
+module LibrariesList
+  class << self
+    def fetch_all_section_projects(user)
+      sections = user.sections + user.sections_as_student
+      projects = []
+      sections.each do |section|
+        projects += ProjectsList.fetch_section_projects(section).select {|project| !project[:libraryName].nil}
```

Comment:
I missed a conceptual piece somewhere - libraries _are_ projects?

Reply:
Ok, so a project is a "library" if it has a libraryName.  This method name would be clearer for me if it was`fetch_section_libraries`. Why fetch all of the projects then filter to only libraries rather than only pull libraries from the pegasus db?

Reply:
Is you security concern that now, theoretically students could access to all of the projects that other students in their sections have made, not just libraries?

Reply:
Also, does a student have to chose to share or publish their library for another student to be able to access it? So many questions - sorry I can wait until in person. Just trying to wrap my head around the feature so I'm better prepared.